tensorly/contrib/decomposition/tt_TTOI.py

```diff
@@ -0,0 +1,255 @@
+    """
+
+    if left_to_right == True:
+        for i in range(len(multiplier_list)):
+            tensor_prod = tl.tensordot(
+                multiplier_list[i], tensor_prod, axes=([0, 1], [0, 1])
+            )
+    else:
+        for i in range(len(multiplier_list)):
+            tensor_prod = tl.tensordot(
+                tensor_prod,
+                multiplier_list[i],
+                axes=([tl.ndim(tensor_prod) - 1, tl.ndim(tensor_prod) - 2], [0, 1]),
+            )
+    return tensor_prod
+
+
+def tensor_train_OI(
+    data_tensor, rank, n_iter=1, trajectory=False, return_errors=True, **context
+):
+    """Perform tensor-train orthogonal iteration (TTOI) for tensor train decomposition
+    Reference paper: Zhou Y, Zhang AR, Zheng L, Wang Y. "Optimal high-order tensor svd via tensor-train orthogonal iteration."
+
+    Parameters
+    ----------
+    data_tensor: tl.tensor
+        observed tensor data
+    rank : tuple
+        rank of the TT decomposition
+        must verify rank[0] == rank[-1] == 1 (boundary conditions)
+        and len(rank) == len(tl.shape(data_tensor))+1
+    n_iter : int
+        half the number of iterations
+    trajectory : bool, optional, default is False
+        if True, the output of each iteration of TTOI is returned: 2*n_iter outputs
+        otherwise, only the output of the last iteration is returned
+    return_errors : bool, optional, default is True
+        if True, the approximation/reconstruction error of each iteration of TTOI is returned: 2*n_iter outputs
+    Returns
+    -------
+    factors : list of n_iter tensors or one tensor
+    * n_iter tensors (if `trajectory` is True) : each list contains the output of each iteration, one full tensor and list of tensor factors
+    * one tensor (otherwise): output of the last iteration, one full tensor and list of tensor factors 
+    full_tensor : list of n_iter tensors or one tensor
+    * n_iter tensors (if `trajectory` is True) : each list contains the output of each iteration, one full tensor and list of tensor factoros
+    * one tensor (otherwise): output of the last iteration, one full tensor and list of tensor factors 
+    """
+    context = tl.context(data_tensor)
+    shape = tl.shape(data_tensor)
+    n_dim = len(shape)
+
+    rank = validate_tt_rank(shape, rank)
+
+    # Make sure it's not a tuple but a list
+    rank = list(rank)
+
+    # Add two one-dimensional mode to data_tensor
+    data_tensor_extended = tl.reshape(data_tensor, (1,) + shape + (1,))
+
+    if trajectory:
+        factors = list()
+        full_tensor = list()
+
+    if return_errors:
+        error_list = list()
+
+    # perform TTOI for n_iter iterations
+    for iteration in range(n_iter):
+        # first perform forward update
+        # left_singular_vectors will be a list including estimated left singular spaces at the current iteration
+        left_singular_vectors = list()
```

Comment:
Thanks for the suggestion! However, our later updates make use of many intermediate product obtained during the TT-SVD algorithm, and it would need a lot of changes in our code structure if we want to use the default TT-SVD function.... If possible, I would prefer not changing this now...